### ISSUE TITLE
cuRoPE2D does not have backward implementation

### ISSUE THREAD

**Issue by sevashasla** [naver/croco#49]
Hello, thank you for your amazing work!
I noticed that cuRoPE2D does not have backward implementation. 
Am I correct that cuRoPE2D should be used only for inference? Do I use cuRoPE2D properly? 

I used a code snippet:
```python3
import torch
from .models.pos_embed import RoPE2D

rope = RoPE2D(100.0)
tokens = torch.randn((1, 2, 3, 4), requires_grad=True).cuda()
tokens.retain_grad()
pos = torch.tensor([[[0, 0], [0, 1], [1, 1]]], dtype=torch.long).reshape(1, 3, 2).cuda()

y = rope(tokens, pos).sum()
y.backward()
tokens.grad
```

Two implementations (kernel and pure pytorch) provide different gradients.

Thank you in advance.
